composer/optim/scheduler.py

```diff
@@ -521,26 +521,63 @@ def __call__(self, state: State, ssr: float = 1.0):
         t_max = _convert_time(self.t_max, state, ssr=ssr)
         current_time = state.timestamp.get(t_max.unit)
         frac_of_total = (current_time / t_max).value
+        frac_of_total = min(1.0, frac_of_total)
```

Comment:
shouldn't we raise error if `frac_of_total > 1.0`? As in, if current time exceeds t_max? This is the slow failing behavior we talked about right

Reply:
Seems like most schedulers account for this behavior already, but it was missing from this one so adding it to make all schedulers consistent

https://github.com/mosaicml/composer/blob/2f11230a17bd35eb4f6a5784a18e43c648a56b2f/composer/optim/scheduler.py#L350
https://github.com/mosaicml/composer/blob/2f11230a17bd35eb4f6a5784a18e43c648a56b2f/composer/optim/scheduler.py#L394
https://github.com/mosaicml/composer/blob/2f11230a17bd35eb4f6a5784a18e43c648a56b2f/composer/optim/scheduler.py#L747
https://github.com/mosaicml/composer/blob/2f11230a17bd35eb4f6a5784a18e43c648a56b2f/composer/optim/scheduler.py#L822
https://github.com/mosaicml/composer/blob/2f11230a17bd35eb4f6a5784a18e43c648a56b2f/composer/optim/scheduler.py#L899